fastparse/shared/src/main/scala/fastparse/sequence/SeqApi.scala

```diff
@@ -0,0 +1,62 @@
+package fastparse.sequence
+
+import scala.reflect.ClassTag
+import fastparse.core
+import fastparse.parsers
+
+/**
+ * This is a trait applicable on potentially any sequences but Vector or Array
+ * are typically usable in real world.
+ */
+abstract class SeqApi[Elem, Repr <: Seq[Elem]](ct: ClassTag[Elem], reprOps: fastparse.utils.ReprOps[Elem, Repr]) {
+  implicit val implicitReprOps = reprOps
+
+  protected[this] implicit val implicitClassTag = ct
+  type ParserInput = fastparse.utils.ParserInput[Elem, Repr]
+  type IndexParserInput = fastparse.utils.IndexedParserInput[Elem, Repr]
+  type IteratorParserInput = fastparse.utils.IteratorParserInput[Elem, Repr]
```

Comment:
Because there is `elemSetHelper: ElemSetHelper[Elem]` in Api.
So I can't convert any object into an Int or toLowerCase.